test/extended/prometheus/prometheus.go

```diff
@@ -159,7 +201,22 @@ var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules [apigro
 				violations.Insert(
 					fmt.Sprintf("WARNING: Alert %q is critical and has no 'runbook_url' annotation", alert.Name),
 				)
-			} else if runbook != "" {
+			}
+
+			return violations
+		})
+
+		if err != nil {
+			e2e.Failf(err.Error())
+		}
+	})
+
+	g.It("should have valid runbook URLs", func() {
+		err := helper.ForEachAlertingRule(alertingRules, func(alert promv1.AlertingRule) sets.String {
+			violations := sets.NewString()
+			runbook := string(alert.Annotations["runbook_url"])
+
+			if runbook != "" {
 				// If there's a 'runbook_url' annotation, make sure it's a
 				// valid URL and that we can fetch the contents.
 				if err := helper.ValidateURL(runbook, 10*time.Second); err != nil {
```

Comment:
I would separate the test to perform 1) the semantic validation of the URL (e.g. it can be parsed by url.Parse() and the scheme is "http" or "https") and  2) the actual verification that the URL returns a 200 response code. The former can fail the test if it detects an invalid URL and the latter should only flake as the target might be temporarily unavailable.

Reply:
The `helper.ValidateURL` function uses `url.Parse(rawURL)` and returns error if it fails to parse.
can i add check for "https" scheme in the same validation function?

https://github.com/openshift/origin/blob/master/test/extended/util/prometheus/helpers.go#L355

Reply:
>  can i add check for "https" scheme in the same validation function?

yes and I would split the actual request to the URL into another function (e.g. QueryURL()).